### ISSUE TITLE
[Mobile Payments] Connecting to a BI reader takes a while

### ISSUE THREAD

**Issue by kidinov** [woocommerce/woocommerce-android#8784]
My Current measurements (2.16b version)

```
TTP PERF:
* Connection token request: ~1300ms
* Location request: ~1000ms
* SDK request to activate a reader : ~2800ms (we send location ID here already, not sure what the SDK makes the second request)
* SDK request to pass again location id : ~600ms
```
In total it's ~8sec, 6 of which is the network requests.

With the reader we fetch location only once, because then a reader keeps it

**Comment by kidinov** [comment]
With the latest 2.19 SDK we have an issue with room version between fluxc and stripe


```
java.lang.NoSuchFieldError: No instance field mDelegate of type Landroid/database/sqlite/SQLiteStatement; in class Landroidx/sqlite/db/framework/FrameworkSQLiteStatement; or its superclasses (declaration of 'androidx.sqlite.db.framework.FrameworkSQLiteStatement' appears in /data/app/~~LN1EFRkGFiTjL67IF0nNVQ==/com.woocommerce.android.dev-AjxIb5jvrXB6u_mn2flNdA==/base.apk!classes20.dex)
```

Room migrated to Kotlin and the `mDelegate` now is `delegate`

**Comment by kidinov** [comment]
With 2.17 the timing is the same. I asked stripe if they can address that p1681199300519189-slack-C01G168NFC2

**Comment by kidinov** [comment]
So far we got answer that on the SDK side they can not/won't do anything about that

**Comment by malinajirka** [comment]
Since there hasn't been any activity since 2023 and the fact that reader connection is usually done just during a session -> closing as `not planned - low priority`.
